File: By-Python/programmers/[위클리챌린지]/상호평가.py
def rotate(scores):
    N, M = len(scores), len(scores[0])
    turned = [[0] * N for _ in range(M)]
    
    for i in range(len(scores)):
        for j in range(len(scores[0])):
            turned[j][i] = scores[i][j]
        
    return turned

def grade(score):
    if score >= 90:
        return "A"
    elif score >= 80:
        return "B"
    elif score >= 70:
        return "C"
    elif score >= 50:
        return "D"
    else:
        return "F"
    
def solution(scores):
    answer = ''    
    tScores = rotate(scores)

    for idx,item in enumerate(tScores):
        if (max(item) == item[idx] or min(item) == item[idx]) and item.count(item[idx]) == 1:
            answer += grade((sum(item) - item[idx])/(len(item)-1))
        else:
            answer += grade(sum(item)/len(item))            
            
    return answer

File: By-Swift/By-Swift/Programmers/위클리챌린지/상호평가.swift
import Foundation

func grade(_ score: Float) -> String {
    switch score {
        case (90...):
            return "A"
        case (80..<90):
            return "B"
        case (70..<80):
            return "C"
        case (50..<70):
            return "D"
        default:
            return "F"
    }
}

func rotate(_ subject: [[Int]]) -> [[Int]] {
    let N = subject.count
    let M = subject.first!.count
    var result = Array(repeating: Array(repeating: 0, count: N), count: M)
    
    for i in 0..<N {
        for j in 0..<M {
            result[j][i] = subject[i][j]
        }
    }
    
    return result
}

func solution(_ scores:[[Int]]) -> String {
    var answer = ""
    let tScore = rotate(scores)
    
    for (idx,item) in tScore.enumerated() {
        var len = item.count
        var sum = item.reduce(0, {$0+$1})
        if (item.max() == item[idx] || item.min() == item[idx]) && item.filter({ $0 == item[idx] }).count == 1
        {
            sum -= item[idx]
            len -= 1
        }
        answer += grade(Float(sum/len))
    }
    
    return answer
}